include/cudaq/Optimizer/Transforms/Passes.td

```diff
@@ -779,6 +779,44 @@ def DeleteStates : Pass<"delete-states", "mlir::ModuleOp"> {
   }];
 }
 
+def StateInitialization : Pass<"state-initialization", "mlir::func::FuncOp"> {
```

Comment:
I think we should tweak this name a bit to be more descriptive. Something more like "replace state with kernel" perhaps?

Reply:
Sounds good, will do